lib/galaxy/datatypes/images.py

```diff
@@ -78,6 +84,34 @@ class Tiff(Image):
     file_ext = "tiff"
 
 
+class OMETiff(Tiff):
+    file_ext = "ome.tiff"
+    MetadataElement(name="offsets", desc="Offsets File", param=FileParameter, file_ext="json", readonly=True, no_value=None, visible=False, optional=True)
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        try:
+            spec_key = 'offsets'
+            offsets_file = dataset.metadata.offsets
+            if not offsets_file:
+                offsets_file = dataset.metadata.spec[spec_key].param.new_file(dataset=dataset)
+            with tifffile.TiffFile(dataset.file_name) as tif:
+                offsets = [page.offset for page in tif.pages]
+            with open(offsets_file.file_name, 'w') as f:
+                json.dump(offsets, f)
+            dataset.metadata.offsets = offsets_file
+        except Exception:
+            pass
```

Comment:
```suggestion
        spec_key = 'offsets'
        offsets_file = dataset.metadata.offsets
        if not offsets_file:
            offsets_file = dataset.metadata.spec[spec_key].param.new_file(dataset=dataset)
        with tifffile.TiffFile(dataset.file_name) as tif:
            offsets = [page.offset for page in tif.pages]
        with open(offsets_file.file_name, 'w') as f:
            json.dump(offsets, f)
        dataset.metadata.offsets = offsets_file
```
I suppose if the metadata setting fails you're not dealing with a valid OMETiff file, and in that case it's better to fail here.